Share button Visibility

Anuj-Rathore24 opened the pull request:
## PR Description 
- This PR adds functionality to control the visibility of a custom share button based on the user's interaction with the Video.js player and the playback state.
- Modifies CSS of the modal for better UI.

### Why this approach
The built-in Video.js control bar visibility logic and related classes (.vjs-control-bar, .vjs-user-active, etc.) do not apply to our custom share button because:

- The share button is rendered outside the standard Video.js control bar.
- Video.js only applies visibility-related classes (vjs-user-active / vjs-user-inactive) to its own elements, not external DOM nodes.

**To test this feature, you can head to my website** - [Live Working](https://anuj-singh.rt.gw/?godam_page=video-preview&id=15)
### Working Example


https://github.com/user-attachments/assets/e98db022-9a86-4911-8c89-c816170d637c



## Screenshot for UI Change
| Before | After |
|--------|-------|
| <img width="400" alt="Screenshot 2025-07-18 at 12 43 16 PM" src="https://github.com/user-attachments/assets/eac5e6d2-00de-458f-9749-8ef7e79c52b5" /> | <img width="400" src="https://github.com/user-attachments/assets/1d4a5e00-3665-4073-9d80-b6897cce44ff" /> |

Resolves: #673

Mukulsingh27 commented:
@Anuj-Rathore24 Looks good!

Here’s some feedback:

- Let’s use CSS to make the share button visible on hover (default it to hidden). It should appear when the user hovers over the video container.
- Also, let’s hide the video controls while the video is playing, and show them when the user hovers over the video.

cc: @prachigarg19 @nayemDevs

Anuj-Rathore24 commented:
Thank you for your feedback @Mukulsingh27, 

I have Some thoughts 🙇 

- Videojs handles visibility for control box using two classes, `vjs-user-inactive ` and `vjs-user-active `. Now since we want to replicate the same for share button, I am using similar method, but cannot use these specific classes as the share button is not added inside videojs. 


REF
https://github.com/videojs/video.js/blob/31f8d7ca102ba1a3bd62ef6b1dc8bd8f6e77719e/src/js/player.js#L4341-L4344 


- Now if I use pure css for this, share buttons functionality won't be replicated as that of control box . 

On the side note, 
- Control box already disappear on users inactivity, as shown in the video, 
https://github.com/user-attachments/assets/9f3cefb1-982f-464d-81db-874f73768303

Anuj-Rathore24 commented:
Do Note, 

While solving linting issue, 

- class `image-cta-btn` has been declared twice, so I have deleted the [older one](https://github.com/rtCamp/godam/blob/556e792c0b8bf83d49debcabcba143fe51d81a33/assets/src/blocks/godam-player/style.scss#L257), [Newest One](https://github.com/rtCamp/godam/blob/556e792c0b8bf83d49debcabcba143fe51d81a33/assets/src/blocks/godam-player/style.scss#L312)
- To resolve `Unexpected unit "em" for property "line-height"` for [line](https://github.com/rtCamp/godam/blob/556e792c0b8bf83d49debcabcba143fe51d81a33/assets/src/blocks/godam-player/style.scss#L474) I have replaced `1.3em` with `1.3`

Mukulsingh27 commented on assets/src/blocks/godam-player/style.scss:
```diff
@@ -621,12 +622,11 @@
 
 		label {
-			color: var(--wp--preset--color--accent-4);
+			color: #52525B;
```
Let's use variables in the _variables.scss file for these custom hex codes. This will help avoid repetition and make future edits easier.

Address everywhere.

Mukulsingh27 commented on assets/src/css/godam-player.scss:
```diff
@@ -835,9 +851,23 @@
-	transition: background-color 0.3s, color 0.3s;
 	border-radius: 50%;
 	display: flex;
+	transition:
+		opacity 0.3s ease,
+		visibility 0.3s ease,
+		background-color 0.3s,
+		color 0.3s;
+}
+
+.godam-share-button:hover{
+	background-color: #AB3A6C;
```
use `transition: all 0.3s ease`

and also use SCSS nesting for hover

Mukulsingh27 commented on assets/src/js/godam-player/frontend.js:
```diff
@@ -530,12 +530,12 @@ function GODAMPlayer( videoRef = null ) {
-						<a class="reddit social-icon" target="blank"><img src=${ Reddit } alt='Reddit icon' height={24} width={24}</a>
-						<a class="whatsapp social-icon" target="blank"><img src=${ Whatsapp } alt='Whatsapp icon' height={24} width={24}</a>
-						<a class="telegram social-icon" target="blank"><img src=${ Telegram } alt='Telegram icon' height={24} width={24}</a>
+						<a class="facebook social-icon" target="blank"><img src=${ Facebook } alt='Facebook icon' height="20" width="20" </a>
+						<a class="twitter social-icon" target="blank"><img src=${ Twitter } alt='Twitter icon' height="20" width="20" </a>
+						<a class="linkedin social-icon" target="blank"><img src=${ LinkedIn } alt='Linkedin icon' height="20" width="20" </a>
+						<a class="reddit social-icon" target="blank"><img src=${ Reddit } alt='Reddit icon' height="20" width="20" </a>
+						<a class="whatsapp social-icon" target="blank"><img src=${ Whatsapp } alt='Whatsapp icon' height="20" width="20" </a>
+						<a class="telegram social-icon" target="blank"><img src=${ Telegram } alt='Telegram icon' height="20" width="20" </a>
```
Also use height and width in SCSS for consistency

```CSS
.share-buttons {
  .social-icon {
    // Set height and width for consistency
    height: 20px;
    width: 20px;
  }
}
```
